src/tagmanager/tm_parser.c

```diff
@@ -44,25 +44,49 @@ typedef struct
 
 static GHashTable *subparser_map = NULL;
 
+#define COMMON_C \
+	{'d', tm_tag_macro_t}, \
+	{'e', tm_tag_enumerator_t}, \
+	{'f', tm_tag_function_t}, \
+	{'g', tm_tag_enum_t}, \
+	{'m', tm_tag_member_t}, \
+	{'p', tm_tag_prototype_t}, \
+	{'s', tm_tag_struct_t}, \
+	{'t', tm_tag_typedef_t}, \
+	{'u', tm_tag_union_t}, \
+	{'v', tm_tag_variable_t}, \
+	{'x', tm_tag_externvar_t},
 
-static TMParserMapEntry map_C[] = {
+/* Old C parser, also used by GLSL and Ferite */
+static TMParserMapEntry map_C_old_parser[] = {
+	COMMON_C
 	{'c', tm_tag_class_t},
-	{'d', tm_tag_macro_t},
-	{'e', tm_tag_enumerator_t},
-	{'f', tm_tag_function_t},
-	{'g', tm_tag_enum_t},
-	{'m', tm_tag_member_t},
 	{'n', tm_tag_namespace_t},
-	{'p', tm_tag_prototype_t},
-	{'s', tm_tag_struct_t},
-	{'t', tm_tag_typedef_t},
-	{'u', tm_tag_union_t},
-	{'v', tm_tag_variable_t},
-	{'x', tm_tag_externvar_t},
 };
 
-/* C++, same as C */
-#define map_CPP map_C
+# define COMMON_C_NEW_PARSER \
+	{'h', tm_tag_undef_t}, \
+	{'l', tm_tag_undef_t}, \
```

Comment:
@elextr If you want to experiment, map the 'l' kind here to tm_tag_variable_t and you'll get all local variables parsed.